### ISSUE TITLE
[FEATURE] Support for wildcard patterns in the replication start , stop APIs

### ISSUE THREAD

**Issue by gbbafna** [opensearch-project/cross-cluster-replication#379]
**Is your feature request related to a problem?**

Several OpenSearch APIs allow the use of wildcard patterns, however this is not supported in the replication APIs (start/stop APIs). Supporting wildcard would allow users to setup replication for several indexes using a single call and the behavior will be consistent with other OpenSearch APIs

**What solution would you like?**
We can start with start and stop APIs. These can be extended to pause and resume APIs as well.

**Comment by gbbafna** [comment]
Related issue : https://github.com/opensearch-project/cross-cluster-replication/issues/344

**Comment by gbbafna** [comment]
We discussed among the team today and the approach we have decided is :

1. Start API : Customers can use Autofollow instead of supplying wildcards. Autofollow gives better retry mechanism, stats APIs and better monitoring overall
2. Stop API : We can support Stop API once it moves to an async operation. Doing it synchronously puts a lot of management headache in case of partial failures, indices count , unresponsive nodes etc and is not worth making the effort for the sync API.
